src/api/app/models/event_subscription/for_role_form.rb

```diff
@@ -0,0 +1,59 @@
+class EventSubscription
+  class ForRoleForm
+    attr_reader :name, :channels, :subscriber
+
+    def initialize(role_name, event, subscriber)
+      @subscriber = subscriber
+      @name = role_name
+      @event = event
+      @channels = []
+    end
+
+    def call
+      @channels = EventSubscription.channels.keys[1..-1].map do |channel|
+        find_subscription_for_event_class_and_role(@event, name, channel)
+      end
+
+      self
+    end
+
+    private
+
+    def find_subscription_for_event_class_and_role(event_class, role, channel)
+      subscriber_subscription =
+        find_subscription_for_subscriber(event_class, role, channel) || find_subscription_for_subscriber(event_class, role, 'disabled')
+      default_subscription = find_default_subscription(event_class, role, channel)
+
+      subscription = if subscriber.present? && subscriber_subscription.present?
```

Comment:
I think you can write something like this
```
subscription = subscriber_subscription if subscriber.present?
subscription ||= default_subscription
subscription ||= EventSubscription.new(subscriber: subscriber,
                                       eventtype: event_class.to_s,
                                       receiver_role: role,
                                       channel: 'disabled')
```
```

Reply:
I will have a look on it

Reply:
I did it in a different way.

Reply:
Okay